rotkehlchen/chain/evm/transactions.py

```diff
@@ -456,19 +457,45 @@ def get_or_query_transaction_receipt(self, tx_hash: EVMTxHash) -> 'EvmTxReceipt'
             )
 
         if len(result) == 0:
-            transaction = self.evm_inquirer.get_transaction_by_hash(tx_hash)
-            with self.database.user_write() as write_cursor:
-                dbevmtx.add_evm_transactions(write_cursor, [transaction], relevant_address=None)
-            period = TimestampOrBlockRange(
-                range_type='blocks',
-                from_value=transaction.block_number,
-                to_value=transaction.block_number,
-            )
-            if transaction.to_address is not None:  # internal transactions only through contracts
-                self._query_and_save_internal_transactions_for_range(
-                    address=None,  # get all internal transactions for the parent hash
-                    period=period,
+            if tx_hash == GENESIS_HASH:
+                # Query genesis transactions
+                with self.database.conn.read_ctx() as cursor:
+                    accounts_data = self.database.get_blockchain_account_data(
+                        cursor,
+                        self.evm_inquirer.chain_id.to_blockchain(),
+                    )
+                for data in accounts_data:
+                    self._query_and_save_transactions_for_range(
+                        address=data.address,
+                        period=TimestampOrBlockRange(range_type='blocks', from_value=0, to_value=0),  # noqa: E501
+                    )
+
+                # Check whether the genesis tx was added
+                with self.database.conn.read_ctx() as cursor:
+                    added_tx = dbevmtx.get_evm_transactions(
+                        cursor=cursor,
+                        filter_=EvmTransactionsFilterQuery.make(tx_hash=GENESIS_HASH, chain_id=self.evm_inquirer.chain_id),  # noqa: E501
+                        has_premium=True,  # we don't need any limiting here
+                    )
+                if len(added_tx) == 0:
+                    raise InputError(
+                        f'There is no tracked {str(self.evm_inquirer.chain_id)} address that '
+                        f'would have a genesis transaction',
+                    )
+            else:
```

Comment:
The else is what was there before, right?

```suggestion
            else:  # normal functionality
```

Reply:
Right